### ISSUE TITLE
stable/prometheus broken on helm release 2.9.0

### ISSUE THREAD

**Issue by georgepapas** [kubernetes/helm#3981]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-12T14:26:04Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-12T14:14:26Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): bare metal

Installing `stable/prometheus` chart is broken on helm `2.9.0`. Works fine in helm `2.8.2`
The prometheus configMap entry contains broken yaml for the `prometheus.yml` key.

Specifically, the `alerting:` section that is added by `templates/server-configmap.yml` is missing a new line and results in the following snippet being generated

```
      - action: replace
        source_labels:
        - __meta_kubernetes_pod_name
        target_label: kubernetes_pod_namealerting:
      alertmanagers:
      - kubernetes_sd_configs:
```

which should actually be like this

```
      - action: replace
        source_labels:
        - __meta_kubernetes_pod_name
        target_label: kubernetes_pod_name
    alerting:
      alertmanagers:
      - kubernetes_sd_configs:
```

Possibly related to #3410 #3470 as referenced in the 2.9.0 release notes

This results in the prometheus server pod failing to starts as it complains about malformed yaml as shown below

```
level=error ts=2018-04-27T04:41:05.546416253Z caller=main.go:582 err="Error loading config couldn't load configuration (--config.file=/etc/config/prometheus.yml): parsing YAML file /etc/config/prometheus.yml: yaml: line 202: mapping values are not allowed in this context"
level=info ts=2018-04-27T04:41:05.546496152Z caller=main.go:584 msg="See you next time!"
```

**Comment by bacongobbler** [comment]
thanks for catching this, @georgepapas. Looks like we'll have to revert that change.

**Comment by ptagr** [comment]
Thanks for fixing this

**Comment by consideRatio** [comment]
#### Explanation of error

The following template code became problematic with #3837.

```yaml
{{- range $key, $value := .Values.serverFiles }}
  {{ $key }}: |
{{ toYaml $value | default "{}" | indent 4 }}
{{- if eq $key "prometheus.yml" -}}
{{- if $root.Values.alertmanager.enabled -}}
    alerting:
```

Note that all whitespace in between the `{{ toYaml ...` output and the `alerting:` line is chomped by the backward and forward comping by lines in between. As #3837 caused `toYaml` to no longer generate an un-chompable new line in the end of `{{ toYaml ...`'s output, `alerting:` is suddenly overlapping on top of the output from `toYaml`.

https://github.com/kubernetes/charts/blob/9667722aba5b1d5e823385db0b3c1d0bfeea2904/stable/prometheus/templates/server-configmap.yaml#L14-L19

---

Sorry for the trouble caused by my PR!

**Comment by bacongobbler** [comment]
This has been fixed in helm v2.9.1, which is available here: https://github.com/kubernetes/helm/releases/v2.9.1

Thanks everyone!
